pontoon/base/models.py

```diff
@@ -2588,7 +2591,22 @@ def map_entities(cls, locale, entities, visible_entities=None):
         for entity in entities:
             translation_array = []
 
-            if entity.string_plural == "":
+            if preferred_source_locale == "":
+                string = entity.string
+                string_plural = entity.string_plural
+            else:
+                preferredSourceLocale = Locale.objects.filter(
+                    code=preferred_source_locale).values_list('pk')[0][0]
+                string = Translation.objects.filter(locale=preferredSourceLocale, entity=entity)
+                if not string:
+                    string = entity.string
+                    string_plural = entity.string_plural
+                else:
+                    string = string.values_list('string')[0]
+                    # replace with Source Locale Translated Plural String
+                    string_plural = entity.string_plural
+
+            if string_plural == "":
```

Comment:
Inside this if-block, you should set the source string to the alternative source locale if available:
```python
if preferred_source_locale != '' and entity.alternative_originals:
    string = entity.alternative_originals[0].string
```

Note that this will only work if you above the `entities` for loop do:

```python
if preferred_source_locale != '':
    entities = entities.prefetch_alternative_originals(preferred_source_locale)
```

With contents of the `prefetch_alternative_originals()` being:

```python
def prefetch_alternative_originals(self, code):
    return self.prefetch_related(
        Prefetch(
            'translation_set',
            queryset=(
                Translation.objects.filter(locale__code=code, approved=True)
            ),
            to_attr='alternative_originals'
        )
    )
```